nutrinator/generator.py

```diff
@@ -0,0 +1,50 @@
+import numpy as np
+from dataclasses import dataclass, field
+from typing import List, Set, Tuple
+
+
+def default_portions() -> List[float]:
+    return [0.25, 0.5, 1, 1.5, 2]
+
+
+def default_taboo() -> Set[int]:
+    return set()
+
+
+@dataclass
+class GeneratorConfig:
+    # If we do the greedy way, we won't need days
+    # days: int = 7
+    dishes: int = 5
+    max_occurence: int = 2
+    recipes_size: int = 100
+    taboo: Set[int] = field(default_factory=default_taboo)
+    portion_set: List[float] = field(default_factory=default_portions)
+
+
+class Generator:
+    def __init__(self, config: GeneratorConfig):
+        self.config = config
+        self.shape = (self.config.dishes,)
+        # self.shape = (self.config.days, self.config.dishes)
+        self.recipes = {i for i in range(config.recipes_size)} - config.taboo
+        self.generator = np.random.default_rng()
+
+    def generate_days(self, size: int, previous: np.ndarray = np.array([])) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
Fajnie byłoby podmienić hinty numpy'owe `np.array` na te z [nptyping](https://stackoverflow.com/questions/54503964/type-hint-for-numpy-ndarray-dtype)